apps/examples/kernel_stability_test/kernel_stability_test_sender.c

```diff
@@ -0,0 +1,202 @@
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <signal.h>
+#include <pthread.h>
+#include <sys/types.h>
+
+#include "kernel_stability_test.h"
+
+static int sig_mq_num;
+static int mq_num;
+
+static void *kernel_stability_test_sig_send(void)
+{
+	int ret;
+	int recv_pid;
+	int pid;
+	mqd_t fd;
+	struct mq_attr attr;
+	char q_name[KERNEL_STABILITY_TEST_MQ_NAME_MAX];
+	union sigval value;
+
+	pid = getpid();
+	printf("[%d] kernel_stability_test_sig_send_pthread \n", pid);
+
+	attr.mq_maxmsg = KERNEL_STABILITY_TEST_MAX_MSG;
+	attr.mq_msgsize = sizeof(int);
+	attr.mq_flags = 0;
+
+	snprintf(q_name, KERNEL_STABILITY_TEST_MQ_NAME_MAX, "%s%d", KERNEL_STABILITY_TEST_SIG_MQ, sig_mq_num++);
+
+	fd = mq_open(q_name, O_RDWR | O_CREAT, 0666, &attr);
+	if (fd < 0) {
+		printf("Failed to open message queue\n");
+		return 0;
+	}
+
+	/* sender receive a receiver's pid. */
+	ret = mq_receive(fd, (char *)&recv_pid, sizeof(int), NULL);
+	if (ret < 0) {
+		printf("Failed to send message\n");
+		mq_close(fd);
+		mq_unlink(q_name);
```

Comment:
In failure case, mq_unlink will be successful only when all the mqueue descriptors are closed.

Reply:
yes. i know. Thanks